doc/source/train/user_guide.rst

```diff
@@ -258,85 +274,77 @@ ready to start training!
 
 .. code-block:: python
 
-    trainer.start() # set up resources
-    trainer.run(train_func)
-    trainer.shutdown() # clean up resources
+    trainer.fit()
 
 Configuring Training
 --------------------
 
 With Ray Train, you can execute a training function (``train_func``) in a
-distributed manner by calling ``trainer.run(train_func)``. To pass arguments
+distributed manner by calling ``Trainer.fit``. To pass arguments
 into the training function, you can expose a single ``config`` dictionary parameter:
 
 .. code-block:: diff
 
     -def train_func():
     +def train_func(config):
 
-Then, you can pass in the config dictionary as an argument to ``Trainer.run``:
+Then, you can pass in the config dictionary as an argument to ``Trainer``:
 
 .. code-block:: diff
 
-    -trainer.run(train_func)
     +config = {} # This should be populated.
-    +trainer.run(train_func, config=config)
+    trainer = TorchTrainer(
+        train_func,
+    +   train_loop_config=config,
+        scaling_config=dict(num_workers=2)
+    )
 
 Putting this all together, you can run your training function with different
 configurations. As an example:
 
 .. code-block:: python
 
-    from ray.train import Trainer
+    from ray.air import session
+    from ray.train.torch import TorchTrainer
 
     def train_func(config):
         results = []
         for i in range(config["num_epochs"]):
-            results.append(i)
-        return results
+            session.report({"epoch": i})
 
-    trainer = Trainer(backend="torch", num_workers=2)
-    trainer.start()
-    print(trainer.run(train_func, config={"num_epochs": 2}))
-    # [[0, 1], [0, 1]]
-    print(trainer.run(train_func, config={"num_epochs": 5}))
-    # [[0, 1, 2, 3, 4], [0, 1, 2, 3, 4]]
-    trainer.shutdown()
+    trainer = TorchTrainer(
+        train_func,
+        train_loop_config={"num_epochs": 2},
+        scaling_config=dict(num_workers=2)
+    )
+    results = trainer.fit()
+    print(results.metrics["num_epochs"])
+    # 1
 
 A primary use-case for ``config`` is to try different hyperparameters. To
 perform hyperparameter tuning with Ray Train, please refer to the
 :ref:`Ray Tune integration <train-tune>`.
 
 .. TODO add support for with_parameters
 
+.. _train-result-object:
 
-.. _train-log-dir:
+The Result Object
```

Comment:
Can we add a TODO here to flesh this section out? Feels like we should have some AIR documentation that describes how to read results